packages/order-watcher/src/order_watcher/order_watcher_websocket.ts

```diff
@@ -0,0 +1,184 @@
+    result: any;
+}
+
+// Wraps the OrderWatcher functionality in a WebSocket server. Motivations:
+// 1) Users can watch orders via non-typescript programs.
+// 2) Better encapsulation so that users can work
+export class OrderWatcherWebSocketServer {
+    public httpServer: http.Server;
+    public readonly _orderWatcher: OrderWatcher;
+    private readonly _connectionStore: Set<WebSocket.connection>;
+    private readonly _wsServer: WebSocket.server;
+    /**
+     * Instantiate a new web socket server which provides OrderWatcher functionality
+     *  @param provider Web3 provider to use for JSON RPC calls (for OrderWatcher)
+     *  @param networkId NetworkId to watch orders on (for OrderWatcher)
+     *  @param contractAddresses Optional contract addresses. Defaults to known
+     *  addresses based on networkId (for OrderWatcher)
+     *  @param partialConfig Optional configurations (for OrderWatcher)
+     */
+    constructor(
+        provider: Provider,
+        networkId: number,
+        contractAddresses?: ContractAddresses,
+        partialConfig?: Partial<OrderWatcherConfig>,
+    ) {
+        this._orderWatcher = new OrderWatcher(provider, networkId, contractAddresses, partialConfig);
+        this._connectionStore = new Set();
+        this.httpServer = http.createServer();
+        this._wsServer = new WebSocket.server({
+            httpServer: this.httpServer,
+            autoAcceptConnections: false,
+        });
+
+        this._wsServer.on('request', (request: any) => {
+            // Designed for usage pattern where client and server are run on the same
+            // machine by the same user. As such, no security checks are in place.
+            const connection: WebSocket.connection = request.accept(null, request.origin);
+            logUtils.log(`${new Date()} [Server] Accepted connection from origin ${request.origin}.`);
+            connection.on('message', this._messageCallback.bind(this, connection));
+            connection.on('close', this._closeCallback.bind(this, connection));
+            this._connectionStore.add(connection);
+        });
+
+        // Have the WebSocket server subscribe to the OrderWatcher to receive updates.
+        // These updates are then broadcast to clients in the _connectionStore.
+        this._orderWatcher.subscribe(this._broadcastCallback);
+    }
+
+    /**
+     * Activates the WebSocket server by having its HTTP server start listening.
+     */
+    public listen(): void {
+        this.httpServer.listen(DEFAULT_HTTP_PORT, () => {
+            logUtils.log(`${new Date()} [Server] Listening on port ${DEFAULT_HTTP_PORT}`);
+        });
+    }
+
+    public close(): void {
+        this.httpServer.close();
+    }
+
+    private _messageCallback(connection: WebSocket.connection, message: any): void {
+        assert.doesConformToSchema('message', message, webSocketUtf8MessageSchema);
+        const requestData: WebSocketRequestData = JSON.parse(message.utf8Data);
```

Comment:
Sorry if my comment wasn't clear but we actually need to validate that `requestData` has the correct structure using JSON schema. We need to ensure that is has the `action` key, and that depending on the action key, that is has the associated/expected `param`. E.g if the `key` is `addOrderAsync`, then the `param` is a `signedOrder`.